src/algokit/core/project/deploy.py

```diff
@@ -4,13 +4,55 @@
 
 import click
 import dotenv
+from algokit_utils import get_algonode_config
 
 from algokit.core.conf import ALGOKIT_CONFIG, get_algokit_config
+from algokit.core.sandbox import (
+    DEFAULT_ALGOD_PORT,
+    DEFAULT_ALGOD_SERVER,
+    DEFAULT_ALGOD_TOKEN,
+    DEFAULT_INDEXER_PORT,
+    DEFAULT_INDEXER_SERVER,
+    DEFAULT_INDEXER_TOKEN,
+)
 from algokit.core.utils import load_env_file, split_command_string
 
 logger = logging.getLogger(__name__)
 
 
+class _KnownEnvironments:
+    LOCALNET = "localnet"
+    MAINNET = "mainnet"
+    TESTNET = "testnet"
+
+
+_mainnet_algod_config = get_algonode_config("mainnet", config="algod", token=DEFAULT_ALGOD_TOKEN)
+_testnet_algod_config = get_algonode_config("testnet", config="algod", token=DEFAULT_ALGOD_TOKEN)
+_mainnet_indexer_config = get_algonode_config("mainnet", config="indexer", token=DEFAULT_INDEXER_TOKEN)
+_testnet_indexer_config = get_algonode_config("testnet", config="indexer", token=DEFAULT_INDEXER_TOKEN)
+
+
+_ENVIRONMENT_CONFIG: dict[str, dict[str, str | None]] = {
+    _KnownEnvironments.LOCALNET: {
+        # this file should contain environment variables specific to algokit localnet
+        "ALGOD_TOKEN": str(DEFAULT_ALGOD_TOKEN),
+        "ALGOD_SERVER": str(DEFAULT_ALGOD_SERVER),
+        "ALGOD_PORT": str(DEFAULT_ALGOD_PORT),
+        "INDEXER_TOKEN": str(DEFAULT_INDEXER_TOKEN),
+        "INDEXER_SERVER": str(DEFAULT_INDEXER_SERVER),
+        "INDEXER_PORT": str(DEFAULT_INDEXER_PORT),
+    },
+    _KnownEnvironments.MAINNET: {
```

Comment:
Do we need ports for the mainnet and testnet configs?

Reply:
+1 - should be 443 right?

Reply:
@robdmoore @neilcampbell i used the values from localnet config from .env.template files we got on smart contract templates. The values are as follows:

ALGOD_TOKEN=aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
ALGOD_SERVER=http://localhost
ALGOD_PORT=4001
INDEXER_TOKEN=aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa
INDEXER_SERVER=http://localhost
INDEXER_PORT=8980

I am pretty sure ports are needed but only on localnet, there is a reason why we got it this way on the template repos. Will double check the behaviour where ports are utilized specifically in localnet cases, i think there is some methods in algokit-utils that rely on existence of those vars in localnet scenarios.